app/src/main/java/com/digitalarchitects/rmc_app/screens/EditMyVehicleScreen.kt

```diff
@@ -0,0 +1,317 @@
+                                viewModel.onEvent(EditMyVehicleUIEvent.SetBrand(it))
+                            },
+                            modifier = Modifier.weight(0.6f)
+                        )
+
+                        RmcTextField(
+                            label = stringResource(id = R.string.year),
+//                            icon = Icons.Filled.Person,
+                            value = uiState.year.toString(),
+                            keyboardOptions = KeyboardOptions.Default.copy(
+                                keyboardType = KeyboardType.Number,
+                                imeAction = ImeAction.Next
+                            ),
+                            onValueChange = {
+                                viewModel.onEvent(EditMyVehicleUIEvent.SetYear(it.toInt()))
+                            },
+                            modifier = Modifier.weight(0.4f)
+                        )
+
+                    }
+
+                    RmcSpacer(8)
+
+                    Row(
+                        modifier = Modifier,
+                        horizontalArrangement = Arrangement.spacedBy(dimensionResource(R.dimen.padding_medium))
+                    ) {
+
+                        RmcTextField(
+                            label = stringResource(id = R.string.model),
+//                            icon = Icons.Filled.Person,
+                            value = uiState.model,
+                            keyboardOptions = KeyboardOptions.Default.copy(
+                                keyboardType = KeyboardType.Text,
+                                imeAction = ImeAction.Next
+                            ),
+                            onValueChange = {
+                                viewModel.onEvent(EditMyVehicleUIEvent.SetModel(it))
+                            },
+                            modifier = Modifier.weight(1f)
+                        )
+                    }
+
+                    RmcSpacer(8)
+
+                    Row(
+                        modifier = Modifier,
+                        horizontalArrangement = Arrangement.spacedBy(dimensionResource(R.dimen.padding_medium))
+                    ) {
+
+                        RmcTextField(
+                            label = stringResource(id = R.string.license_plate),
+//                            icon = Icons.Filled.Person,
+                            value = uiState.licensePlate,
+                            keyboardOptions = KeyboardOptions.Default.copy(
+                                keyboardType = KeyboardType.Text,
+                                imeAction = ImeAction.Next
+                            ),
+                            onValueChange = {
+                                viewModel.onEvent(EditMyVehicleUIEvent.SetLicensePlate(it))
+                            },
+                            modifier = Modifier.weight(1f)
+                        )
+
+                        RmcTextField(
```

Comment:
- Als je hier in typt crashed de app. Even kijken hoe we dit oplossen/afvangen?
- 0.0 blijft altijd staan in het veld als je typt

Reply:
Dan gaat er iets fout met de states, ik ga even kijken.